### ISSUE TITLE
LTTP Co-op issues with keys (even jar/monster key drops)

### ISSUE THREAD

**Issue by kngrace** [TestRunnerSRL/bizhawk-co-op#35]
tl;dr Keys obtained from jars and critters sometimes do not appear. These are the keys that are not supposed to be randomized. The only co-op setting we had toggled on was "Raid Boss."

Today I played my first lttp randomizer with a buddy and decided to do co-op. We set up 1.12 Bizhawk and everything looked great. The ONLY option we selected was "Raid Bosses"

Very early on we were a bit confused because my buddy got key-locked out of completing both the rescue Zelda sequence and the sewer sequence later (when we eventually got bracers and he could push the bookcase out of the way to reveal the sewers). 

Over and over again I got key-locked out of dungeons which was annoying because we did raid boss HP with the intent to fight the bosses together. We just figured it was something to do with the co-op logic and worried less when the other one of us was able to finish the dungeon. 

But then we finally hit a crystal dungeon (swamp) that BOTH of us were key-locked out of and a very observant viewer on twitch noticed something was very very wrong. Jars were not holding their expected keys for either of us. We finally realized we were hard locked out of completing the game and decided to completely shut down and restart bizhawk and the co-op script to see if that would fix anything. 

We went back into the swamp and my buddy's key did NOT drop from the jar it was supposed to be in, but suddenly my previously missing key did. I had him re-try the Zelda rescue and sewer escape areas and he was able to find the previously missing keys and full clear them. I then got stuck in sewer again and my viewer said there was a jar I opened that should have had a key but didn't.

So basically, the keys confirmed to glitch out and not appear on some jars and from at least one critter in the sewers. A full reset seemed to make some of the missing keys be obtainable but keys started being unobtainable again shortly after restart. I can't find a full log of how the script split our items up to confirm things and the randomizer spoiler log also won't say because it doesn't know it was created for co-op. Otherwise I am happy to share whatever information you may deem useful to troubleshoot.

For what its worth, my buddy and I have done multiple multi-worlds in Ocarina of Time with no major issues. This was our first time in lttp co-op and thus trust this script more than the other one that is available since we're already familiar with it. We'd like to keep using it if it works properly!

**Comment by kngrace** [comment]
Looking through other issues, the key issue was briefly mentioned in the issue posted by @singularity125 titled [Lttp Co-op Not splitting items across players](https://github.com/TestRunnerSRL/bizhawk-co-op/issues/9)
